src/btree/bt_cursor.c

```diff
@@ -690,7 +690,7 @@ __wt_btcur_search_near(WT_CURSOR_BTREE *cbt, int *exactp)
          * here because at low isolation levels, new records could appear as we are stepping through
          * the tree.
          */
-        while ((ret = __wt_btcur_next(cbt, false)) != WT_NOTFOUND) {
+        while ((ret = __wt_btcur_next(cbt, false, &state.key)) != WT_NOTFOUND) {
```

Comment:
This is a little subtle - the prefix needs to be the key that the user sets via set_key prior to the search_near, right?

There isn't another key that's considered the prefix for the search as well?

Reply:
Correct its the key passed by the user.